src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -439,12 +439,14 @@ void RegionNode::set_loop_status(RegionNode::LoopStatus status) {
   _loop_status = status;
 }
 
-#ifdef ASSERT
-void RegionNode::verify_can_be_irreducible_entry() const {
-  assert(loop_status() == RegionNode::LoopStatus::MaybeIrreducibleEntry, "must be marked irreducible");
-  assert(!is_Loop(), "LoopNode cannot be irreducible loop entry");
+// A Region can only be an irreducible entry if:
+// - It is marked as "maybe irreducible entry". Any other loop status would guarantee
+//   that it is never an irreducible loop entry.
+// - It is not a LoopNode, those are guaranteed to be reducible loop entries.
```

Comment:
Do you mean "**And** it is not a Loop node" (based on condition code below)?
Do we have an assert which checks that `LoopNode` is never marked `MaybeIrreducibleEntry`?

Reply:
>Do you mean "And it is not a Loop node" (based on condition code below)?
I changed it to an "And" now.

>Do we have an assert which checks that LoopNode is never marked MaybeIrreducibleEntry?
I added an assert for `set_loop_status`, just in case.